### ISSUE TITLE
[W3C] Add Distributed tracing demo app to illustrate API usage

### ISSUE THREAD

**Issue by jasonjkeller** [newrelic/newrelic-java-agent#172]
This work is dependent on #169 providing a general use API for creating/accepting distributed tracing payloads.

This is not a hard requirement but as a form of "enhanced documentation" it might be helpful to provide customers with a project that demos how to use our new distributed trace APIs in a project that has microservices communicating via W3C trace context.

We may be able to modify [spring-petclinic-microservices](https://github.com/spring-petclinic/spring-petclinic-microservices) for such purposes.

This is a "nice to have" so depending on scope we may choose to drop this work.

---

Success criteria:
* Customers can clone a fully working project that demos distributed tracing API usage

**Comment by jasonjkeller** [comment]
Demo app: https://github.com/jasonjkeller/kafka-examples
